### ISSUE TITLE
Exclude Global Prefix + Middleware leads to wrong params

### ISSUE THREAD

**Issue by mlbonniec** [nestjs/nest#13318]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When using global prefix, excluded routes still get params in middlewares.

I'm using a `/:instance` global prefix.
I don't want `/stripe/callback` to be affected by the prefix, so I added it in the `exclude` property.
However, my middleware, which is applied to all (`*`) routes still get a `req.params.instance` with the first part of the URL.

I know that I could exclude the middleware with the `.exclude(...)` method, but it will lead to code duplication :(

> [!NOTE]
> Using URL versioning leads to middleware `exclude` not working… is it normal ?

### Minimum reproduction code

https://codesandbox.io/p/devbox/global-prefix-middleware-3ngvgd

### Steps to reproduce

1. `npm run start`
2. Go to https://3ngvgd-3000.csb.app/stripe/callback. You should see "An instance has been found 'stripe'"

### Expected behavior

`req.params.instance` should be `undefined` for path applied in the `setGlobalPrefix` `exclude` property.

### Package

- [ ] I don't know. Or some 3rd-party package
- [X] <code>@nestjs/common</code>
- [X] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

10.3.2

### Packages versions

```bash
[System Information]
OS Version     : Linux 6.1.43
NodeJS Version : v20.9.0
NPM Version    : 9.8.1 

[Nest CLI]
Nest CLI Version : 10.3.1 

[Nest Platform Information]
platform-express version : 10.3.2
schematics version       : 10.1.0
testing version          : 10.3.2
common version           : 10.3.2
core version             : 10.3.2
cli version              : 10.3.1

[Warnings]
The following packages are not in the same minor version
This could lead to runtime errors
* Under version 10.3
- @nestjs/platform-express 10.3.2
- @nestjs/common 10.3.2
- @nestjs/core 10.3.2
* Under version 10.1
- @nestjs/schematics 10.1.0
```

### Node.js version

20.9.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [X] Linux

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR for this issue?

**Comment by kamilmysliwiec** [comment]
This should be fixed in 10.3.4
